core/bin/external_node/src/main.rs

```diff
@@ -954,6 +953,16 @@ async fn run_node(
     )
     .await?;
     let sigint_receiver = env.setup_sigint_handler();
+    // Spawn reacting to signals in a separate task so that the node is responsive to signals right away
+    // (e.g., during the initial reorg detection).
+    tokio::spawn({
+        let stop_sender = stop_sender.clone();
+        async move {
```

Comment:
how about adding it to the task set? It would eliminate tokio::select entirely.

Reply:
This is clever! However, [logging in `ManagedTasks`](https://github.com/matter-labs/zksync-era/blob/76564c54777e244925798952257107d8b9d76bfc/core/lib/utils/src/wait_for_tasks.rs#L37) assumes that a task terminating is an error. I don't really want to change `ManagedTasks` to account for this additional task given that we'll (hopefully) switch to the node framework in the midterm.